### ISSUE TITLE
fork fork chit chat

### ISSUE THREAD

**Issue by yrmvgh** [bleakley/brogue-plus#1]
I have the brogue source on my machine, and I've made some changes to it that I _think_ would interesting, or would contribute to making the game more consistent and less frustrating, however I suck at ... a lot of things. So I have yet to figure out how to compile it and test the changes out. If I run a diff, would you be interested in reviewing the changes and possibly adding them to your fork?

**Comment by yrmvgh** [comment]
> more consistent and less frustrating

For instance, the pure-bad potions have a beneficial side effect that, while small, makes it theoretically possible that a situation exists in which you might drink it rather than just discarding it.

**Comment by yrmvgh** [comment]
``` c
        case POTION_HALLUCINATION:
            player.status[STATUS_HALLUCINATING] = player.maxStatus[STATUS_HALLUCINATING] = 300;
            makePlayerTelepathic(2);
            message("colors are everywhere! The walls are singing! Sight beyond sight!", false);
            break;
```

**Comment by yrmvgh** [comment]
this one is pretty inefficiently coded; my apologies :grinning: 

``` c
        case POTION_DARKNESS:
            hadEffect = false;
            hadEffect2 = false;
            for (tempItem = packItems->nextItem; tempItem != NULL; tempItem = tempItem->nextItem) {
                if (tempItem->category & CAN_BE_DETECTED) {
                    detectMagicOnItem(tempItem);
                    if (itemMagicChar(tempItem)) {
                        if (tempItem->flags & ITEM_MAGIC_DETECTED) {
                            hadEffect2 = true;
                        }
                    }
                }
            }
            if (hadEffect || hadEffect2) {
                if (hadEffect && hadEffect2) {
                    message("you can somehow feel the presence of magic on the level and in your pack.", false);
                } else if (hadEffect) {
                    message("you can somehow feel the presence of magic on the level.", false);
                } else {
                    message("though blinded to the outside world, you can somehow feel the presence of magic in your pack.", false);
                }
            } else {
                message("you can somehow feel the absence of magic in your pack.", false);
            }
            player.status[STATUS_DARKNESS] = max(400, player.status[STATUS_DARKNESS]);
            player.maxStatus[STATUS_DARKNESS] = max(400, player.maxStatus[STATUS_DARKNESS]);
            updateMinersLightRadius();
            updateVision(true);
            message("your vision flickers as a cloak of darkness settles around you!", false);
            break;
```

**Comment by bleakley** [comment]
Hi yrmvgh, thanks for the suggestion.

What platform are you using to compile? If you use Windows with Codeblocks it is somewhat more straightforward. (I haven't compiled it with any other IDE). I could help you out with it.

In this mod there is already a very minor and subtle use of hallucination- a medusa gives you a "petrifying" status as soon as you can see her, which increases as long as she stays in your line of sight. Either stepping out of LOS or drinking a hallucination potion (so you can no longer tell she is a medusa) will stop the petrification.

I had the same feeling, that every item should have _some_ beneficial use, however minor. Medusas do not occur in every seed however, so this item is still useless the majority of the time.

I'm not sure if I want to merge new changes in right away, I actually want to revert some of my changes (like the whip effect) that didn't really work out. I haven't really worked on Brogue+ for over a year; I just recently moved it to github to help organize my past coding projects.

In general I am open to new ideas, but this project is on the back burner at the moment.

**Comment by yrmvgh** [comment]
Okay. If it's on the back burner, I think I will make my own fork and mess around with it, rather than spamming you with PRs when you're not actively working on this. But I'll be using your fork as a base because I love the changes! :grinning: 

As for the compiling issues, I am using a debian-based Linux. I know, I've made my life harder this way, but Windows was starting to annoy me on an almost daily basis, so I evicted it. I have less features and less knowledge of its ins and outs, but nothing is changing itself without my say-so anymore. :wink: 

I will read Brogue's documentation and attempt compiling a few more times. I can (and do) simply run Brogue+ in Wine, but I'd rather try to get it running in Linux like vanilla Brogue does.

But if you ever want to collab, hit me up!

**Comment by bleakley** [comment]
Sounds great! I will definitely get back into Brogue modding if 1.7.5 is ever released. Maybe we could work together on a new mod then. In the meantime I will keep an eye on your fork and check out your new changes.

**Comment by yrmvgh** [comment]
Do you recall off the top of your head what some changes are that I should consider rolling back, other than whips?

**Comment by yrmvgh** [comment]
Don't watch with too much excitement, I'm helping on a DCSS fork right now, and not quite pulling my weight yet. Then again brogue was my first love! :wine_glass:

**Comment by bleakley** [comment]
I think there should be some sort of change to whips- Powerful runic whips are a lot of fun, but it doesn't make sense that loudest weapon in the real world would be the best stealth weapon in Brogue.

The aggravate monsters effect was just too much though- Maybe the player's stealth radius should increase by the whips attack range every time he or she makes an attack?

**Comment by bleakley** [comment]
I'm curious, what is your opinion on goblin skirmishers? I like the general concept but they aren't as much fun as I'd hoped.

**Comment by yrmvgh** [comment]
They provide a sense of novelty on your first few runs! That counts for something, don't it?

**Comment by yrmvgh** [comment]
I haven't tried managing them as a pet, though. That might be cool.

**Comment by bleakley** [comment]
That's true, if it wasn't for goblin skirmishers most games at low depths would look just like vanilla Brogue.

**Comment by yrmvgh** [comment]
Do you have any idea why my "messing around" folder has many more files than my "base" folder? It's been long enough that I have no idea how they got there. Your repository doesn't appear to have these files either:

``` c
diff ~/Desktop/brog/src/brogue/ ~/Desktop/brogue-1.7.4/src/brogue/
Only in ~/Desktop/brogue-1.7.4/src/brogue/: Architect.o
Only in ~/Desktop/brogue-1.7.4/src/brogue/: Buttons.o
Only in ~/Desktop/brogue-1.7.4/src/brogue/: Combat.o
Only in ~/Desktop/brogue-1.7.4/src/brogue/: decss2.pl
Only in ~/Desktop/brogue-1.7.4/src/brogue/: Dijkstra.o
diff ~/Desktop/brog/src/brogue/Globals.c ~/Desktop/brogue-1.7.4/src/brogue/Globals.c
1702c1702
<       "poking at", "Examining",
---
>       "poking at", "Staring blankly",
Only in ~/Desktop/brogue-1.7.4/src/brogue/: Globals.o
Only in ~/Desktop/brogue-1.7.4/src/brogue/: Grid.o
Only in ~/Desktop/brogue-1.7.4/src/brogue/: IO.o
etceteraetcetera, the list is about double this size
```

**Comment by yrmvgh** [comment]
Thinking of increasing the blood spatter amount and shifting the terrain colors, for more of that "things are different" feel like the skirmisher gives

**Comment by yrmvgh** [comment]
Firebolt has a "red" value of 500 - it looks like 100 is the normal cap, does the extra amount of color "spill" out into adjacent cells?

**Comment by bleakley** [comment]
The .o files are the compiled .c files, right? You shouldn't need them. You only need to edit and build the .c and .h files.

**Comment by bleakley** [comment]
There was an interesting idea brought up on the old forums- Salamanders and Naga could retreat into lava or water, respectively, when wounded and they would regenerate quickly while submerged.

**Comment by yrmvgh** [comment]
That's probably well beyond my skills. ;) I don't actually know C, I've just learned to recognize structs and switches and while loops, so that I can change the values a bit.

**Comment by yrmvgh** [comment]
The travis-ci thing on github might be able to do some debugging though, which'll be handy!

**Comment by yrmvgh** [comment]
Whatcha think about bog bodies being able to walk in deep water, like nagas? That seems more intuitive to me, but might annoy people who've learned that they can escape in deep water.

**Comment by yrmvgh** [comment]
On that note, losing items in deep water seems more like a simulationist thing that doesn't make for good gameplay. But maybe it helps teach "large, deep pools are not a place of happiness"?

**Comment by bleakley** [comment]
Here's an annoying 1.7.4 bug that I hadn't heard of until now:
https://www.reddit.com/r/brogueforum/comments/4om2vo/y_u_troll_me_commutation/
